staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/cel/compilation.go

```diff
@@ -160,8 +166,42 @@ func prepareEnvSet(baseEnvSet *environment.EnvSet, declType *apiservercel.DeclTy
 			EnvOptions: []cel.EnvOption{
 				cel.Variable(OldScopedVarName, scopedType.CelType()),
 			},
+			DeclTypes: []*apiservercel.DeclType{
+				scopedType,
+			},
 		},
 	)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	optionalOldSelfEnvSet, err := baseEnvSet.Extend(
+		environment.VersionedOptions{
+			// Feature epoch was actually 1.23, but we artificially set it to 1.0 because these
+			// options should always be present.
+			IntroducedVersion: version.MajorMinor(1, 0),
+			EnvOptions: []cel.EnvOption{
+				cel.Variable(ScopedVarName, scopedType.CelType()),
+			},
+			DeclTypes: []*apiservercel.DeclType{
+				scopedType,
+			},
+		},
+		environment.VersionedOptions{
+			IntroducedVersion: version.MajorMinor(1, 24),
```

Comment:
Wondering if/how versioning/rollout is affected by this variable decl cc @jpbetz

Reply:
use of this env is gated by an actual API field with an actual feature gate attached, so I don't think this is really relevant. I think we should match the existing `oldSelf` initial version here